contracts/examples/lottery-esdt/src/lottery.rs

```diff
@@ -182,15 +162,30 @@ pub trait Lottery {
     }
 
     #[endpoint]
-    fn determine_winner(&self, lottery_name: ManagedBuffer) {
+    fn determine_winner(&self, lottery_name: ManagedBuffer) -> AwardingStatus {
+        let sc_address = self.blockchain().get_sc_address();
+        let sc_address_shard = self.blockchain().get_shard_of_address(&sc_address);
+        let caller = self.blockchain().get_caller();
+        let caller_shard = self.blockchain().get_shard_of_address(&caller);
+        // require!(
+        //     sc_address_shard != caller_shard,
+        //     "Caller needs to be on a remote shard"
+        // );
+
         match self.status(&lottery_name) {
             Status::Inactive => sc_panic!("Lottery is inactive!"),
             Status::Running => sc_panic!("Lottery is still running!"),
             Status::Ended => {
-                self.distribute_prizes(&lottery_name);
-                self.clear_storage(&lottery_name);
+                if self.total_winning_tickets(&lottery_name).is_empty() {
```

Comment:
Maybe add a check for `lottery_info(&lottery_name).is_empty()`. Sure, for now it will crash anyway when you try a `get()`, but it's safer this way.